editor/src/components/inspector/sections/component-section/property-control-controls.tsx

```diff
@@ -437,28 +439,68 @@ const NumberWithSliderControl = React.memo(
   },
 )
 
+export const ImagePreviewTestId = 'image-preview'
+
 export const StringInputPropertyControl = React.memo(
   (props: ControlForPropProps<StringInputControlDescription>) => {
     const { propName, propMetadata, controlDescription } = props
 
     const controlId = `${propName}-string-input-property-control`
     const value = propMetadata.propertyStatus.set ? propMetadata.value : undefined
 
+    const safeValue = typeof value === 'string' ? value : ''
+
     return (
-      <StringControl
-        key={controlId}
-        id={controlId}
-        testId={controlId}
-        value={value ?? ''}
-        onSubmitValue={propMetadata.onSubmitValue}
-        controlStatus={propMetadata.controlStatus}
-        controlStyles={propMetadata.controlStyles}
-        focus={props.focusOnMount}
-      />
+      <FlexColumn style={{ gap: 5 }}>
+        <StringControl
+          key={controlId}
+          id={controlId}
+          testId={controlId}
+          value={safeValue}
+          onSubmitValue={propMetadata.onSubmitValue}
+          controlStatus={propMetadata.controlStatus}
+          controlStyles={propMetadata.controlStyles}
+          focus={props.focusOnMount}
+        />
+        {when(isImageUrl(safeValue), <ImagePreview url={safeValue} />)}
+      </FlexColumn>
     )
   },
 )
 
+interface ImagePreviewProps {
+  url: string
+}
+const ImagePreview = React.memo((props: ImagePreviewProps) => {
+  const [imageCanBeLoaded, setImageCanBeLoaded] = React.useState(true)
```

Comment:
won't that always start with `true` (even if the url hadn't changed) and so it will always first try to the render the `<img>` on every re-render (even if we know the url is bad)?

Reply:
it would start with true on every re-mount, but not on every re-render

Reply:
What Balazs wrote and when the url is bad we don't even mount the ImagePreview (see line 465)
But I can integrate that check into the ImagePreview component itself (by initializing imageCanBeLoaded with `isImageUrl(url)`)

Reply:
yes, of course, re-mount was what I meant 🤦‍♂️ in any case it's not critical at all